source/misc_utils_test.go

```diff
@@ -0,0 +1,257 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package source_test
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/SpeakData/minimarketo"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/goombaio/namegenerator"
+	"github.com/rustiever/conduit-connector-marketo/config"
+	"github.com/rustiever/conduit-connector-marketo/source"
+	sourceConfig "github.com/rustiever/conduit-connector-marketo/source/config"
+)
+
+// actionTypes for createOrUpdate API endpoint
+const (
+	CreateOnly = "createOnly"
+	UpdateOnly = "updateOnly"
+)
+
+var (
+	ClinetID       = os.Getenv("MARKETO_CLIENT_ID")
+	ClientSecret   = os.Getenv("MARKETO_CLIENT_SECRET")
+	ClientEndpoint = os.Getenv("MARKETO_CLIENT_ENDPOINT")
+	Seed           = time.Now().UTC().UnixNano()
+	Fields         = []string{"firstName", "lastName", "email", "createdAt", "updatedAt"} // fields to be returned by the API
+)
+
+// custom wrapper client for minimarketo client
+type Client struct {
```

Comment:
What's the difference between this one, and the "real" one in `source`?

Reply:
Actually no diffrence, both are same, But this client contains API methods for testing purpose.